defs.bzl

```diff
@@ -104,7 +147,7 @@ def typescript_proto_library_aspect_(target, ctx):
 
```

Comment:
So apparently this doesn't behave as I thought it did. I expected the aspect to use the dependency graph to build the correct outputs. However, this doesn't actually declare the outputs for any of the dependencies. It seems like the for loop should be:

`for src in depset(target.proto.direct_sources) + target.proto.transitive_sources:`

So that it actually builds the union of all of the direct sources and any dependencies. Otherwise, you'll get import references to files that don't end up getting copied over to bazel-bin. @Dig-Doug, did you intend for this?

Note - this is another way to expose the problem I encountered with com_google_protobuf. As soon as one of the transitive sources is in a different package, it will break. I'm wondering if what I'm trying to do here is even possible. I was under the impression that it was, but now I'm not sure how to accomplish it.

Reply:
Missed this before. You're correct that we want to generate the files for the transitive dependencies. This way this works is the aspect is run by bazel on each node in the dependency graph. What I do is I pass the generated outputs up from the direct dependencies [here](https://github.com/improbable-eng/ts-protoc-gen/blob/master/defs.bzl#L130), so all the transitive dependencies bubble up to the root (in `dep_js`).

Is the problem you mentioned fixed now?